libbeat/publisher/queue/diskqueue/queue.go

```diff
@@ -281,6 +299,23 @@ func (dq *diskQueue) Producer(cfg queue.ProducerConfig) queue.Producer {
 	}
 }
 
+// Metrics returns current disk metrics
 func (dq *diskQueue) Metrics() (queue.Metrics, error) {
-	return queue.Metrics{}, queue.ErrMetricsNotImplemented
+	respChan := make(chan metricsRequestResponse, 1)
+	req := metricsRequest{response: respChan}
+
+	select {
+	case <-dq.done:
+		return queue.Metrics{}, io.EOF
+	case dq.metricsRequestChan <- req:
+	}
+	resp := <-respChan
+
+	maxSize := dq.settings.MaxBufferSize
+	return queue.Metrics{
+		ByteLimit:            opt.UintWith(maxSize),
+		ByteCount:            opt.UintWith(resp.sizeOnDisk),
+		UnackedConsumedBytes: opt.UintWith(resp.OccupiedRead),
```

Comment:
```suggestion
		UnackedConsumedBytes: opt.UintWith(resp.UnackedConsumedBytes),
```

Can we use the same name for the internal and external metrics?